eslint/babel-eslint-parser/src/convert/convertTokens.cts

```diff
@@ -84,16 +85,33 @@ function convertTemplateType(tokens, tl) {
   return result;
 }
 
-function convertToken(token, source, tl) {
+function convertToken(
+  token: BabelToken,
+  source: string,
+  tl: Record<string, any>,
+) {
   const { type } = token;
   const { label } = type;
-  token.range = [token.start, token.end];
+
+  const newToken: {
+    type: string;
+    range?: [number, number];
+    value?: string;
+    regex?: {
+      pattern: string;
+      flags: string;
+    };
+  } = {
+    ...token,
+    type: "",
+    range: [token.start, token.end],
+  };
```

Comment:
```suggestion
  const newToken: {
    type: string;
    range?: [number, number];
    value?: string;
    regex?: {
      pattern: string;
      flags: string;
    };
  } = token as any;
  newToken.range = [token.start, token.end];
```

Could we do this instead of changing the code to create a new object?